processing/src/main/java/org/apache/druid/java/util/metrics/SysMonitorOshi.java

```diff
@@ -0,0 +1,407 @@
+  {
+    for (Stats stats : statsList) {
+      stats.emit(emitter);
+    }
+    return true;
+  }
+
+  private interface Stats
+  {
+    void emit(ServiceEmitter emitter);
+  }
+
+  private class MemStats implements Stats
+  {
+    @Override
+    public void emit(ServiceEmitter emitter)
+    {
+      GlobalMemory mem = hal.getMemory();
+      if (mem != null) {
+        final Map<String, Long> stats = ImmutableMap.of(
+            "sysO/mem/max", mem.getTotal(),
+            "sysO/mem/used", mem.getTotal() - mem.getAvailable(),
+            "sysO/mem/free", mem.getAvailable()
+        );
+        final ServiceMetricEvent.Builder builder = builder();
+        MonitorUtils.addDimensionsToBuilder(builder, dimensions);
+        for (Map.Entry<String, Long> entry : stats.entrySet()) {
+          emitter.emit(builder.build(entry.getKey(), entry.getValue()));
+        }
+      }
+    }
+  }
+
+  private class SwapStats implements Stats
+  {
+    private long prevPageIn = 0;
+    private long prevPageOut = 0;
+
+    @Override
+    public void emit(ServiceEmitter emitter)
+    {
+      VirtualMemory swap = hal.getMemory().getVirtualMemory();
+
+      if (swap != null) {
+        long currPageIn = swap.getSwapPagesIn();
+        long currPageOut = swap.getSwapPagesOut();
+        final Map<String, Long> stats = ImmutableMap.of(
+            "sysO/swap/pageIn", currPageIn - prevPageIn,
+            "sysO/swap/pageOut", currPageOut - prevPageOut,
+            "sysO/swap/max", swap.getSwapTotal(),
+            "sysO/swap/free", swap.getSwapTotal() - swap.getSwapUsed()
+        );
+
+        final ServiceMetricEvent.Builder builder = builder();
+        MonitorUtils.addDimensionsToBuilder(builder, dimensions);
+        for (Map.Entry<String, Long> entry : stats.entrySet()) {
+          emitter.emit(builder.build(entry.getKey(), entry.getValue()));
+        }
+
+        this.prevPageIn = currPageIn;
+        this.prevPageOut = currPageOut;
+      }
+    }
+  }
+
+  private class FsStats implements Stats
+  {
+    @Override
+    public void emit(ServiceEmitter emitter)
+    {
+      FileSystem fileSystem = os.getFileSystem();
+      for (OSFileStore fs : fileSystem.getFileStores()) {
+        final String name = fs.getName();
+        if (fsTypeWhitelist.contains(fs.getType())) {
+          final Map<String, Long> stats = ImmutableMap.<String, Long>builder()
+                                                      .put("sysO/fs/max", fs.getTotalSpace())
+                                                      .put("sysO/fs/used", fs.getUsableSpace())
+                                                      .put("sysO/fs/files/count", fs.getTotalInodes())
+                                                      .put("sysO/fs/files/free", fs.getFreeInodes())
+                                                      .build();
+          final ServiceMetricEvent.Builder builder = builder()
+              .setDimension("fsDevName", fs.getName())
```

Comment:
I think that volume name should be used for fsDevName and Name for the dirName dimension. thoughts?

Reply:
I think `fs.getName()` returns volume name only. We're emitting 160 metrics in OshiSysMonitor, Let me share them on slack and you can verify if there's some issue. That would be easier

Reply:
@abhishekagarwal87 Marking it unresolved until we figure out best way to get fsTypeName.